### DISCUSSION TITLE
Is it possible to use a sidebar navigation in a GFM document?

### DISCUSSION THREAD

**Discussion by blechturm** [quarto-dev/quarto-cli#7336] [Q&A] [answered]
### Description

I am generating a dynamic quarto document for our database documentation which is going to be hosted in a gitlab repo in the form of a GFM markdown.

Since the document will contain loads of headings and subheadings (each table/view will get a heading) a sidebar navigation similar to the HTML output would be super useful. 

Does anyone know if that is doable in GFM? 

A "Back" button/link for each section which jumps top the TOC would do the trick too I guess.

**Comment by mcanouil** [comment]
GFM is "regular" markdown.
There is no such thing as sidebar, tabsets, or basically HTML structure (with few exception such as tables and figures).

You can reference any headings, so you can just do that at the bottom of your document: `[Back to top](#first-heading)`.

`toc: true` will add a TOC at the beginning of the document.

See <https://quarto.org/docs/reference/formats/markdown/gfm.html>.

**Reply by blechturm** [reply]
I see, is it possible to create a heading for the TOC so I can abuse the `[Back to top](#first-heading)` trick to make the navigation in the document easier?

**Reply by mcanouil** [reply, marked as answer]
Do what you want, note that `#readme` anchor in GitHub means the beginning of the `README.md` file.
Also GitHub creates a "sidebar":
<img width="761" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/4b96e0fc-5d39-421e-8a2a-0a7d41c35cef">

Maybe there are similar (likely not identical) features in GitLab.

For the back to top link, simply targets the title.
See "Heading identifiers" in Pandoc documentation.

````qmd
---
title: "Quarto Playground"
format: gfm
toc: true
---

## Title 1

{{< lipsum 5 >}}

### Title 2

This is a playground for Quarto.

[back to top](#quarto-playground)
````

**Reply by blechturm** [reply]
Sheesh... I totally did not see that dropdown button... This solves everything.

`[back to top](#quarto-playground)` also worked like a charm. 

Thanks a lot 👍
